### PR TITLE
Pause game

### PR DESCRIPTION
**Pull request by LeoWikstrom**
Press ESC to pause game. Press ESC again to unpause game. Press End to quit game. Buttons on the pause menu is another task. Removed unnecessary warning from several NavMeshes. Fixed "Set scene" warning. Made sure all maps are loaded in LoadScene, removing the occasionally long loading time when pressing Start. Fixed skybox to make sure the moon is where the light comes from. I think that was all

### CODE REVIEW

File: StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp [hunk 1]
```diff
@@ -79,3 +82,37 @@ void AudioVoice::Stop()
...
     }
 }
+
+void AudioVoice::Resume()
+{
+    if (m_Initialized)
+    {
+        // reset the buffer if sound has ended
+        XAUDIO2_VOICE_STATE test;
+        m_pSourceVoice->GetState(&test);
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp]: Please change the variable name from the temporary "test" to something more befitting. voiceState perhaps?

File: StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp [hunk 2]
```diff
@@ -79,3 +82,37 @@ void AudioVoice::Stop()
...
+            // reset the buffer so the sound starts from the beginning at next playback
+            m_pSourceVoice->FlushSourceBuffers();
+            m_pSourceVoice->SubmitSourceBuffer(m_pBuffer);
+        }
+
+        // play the sound
+        if (FAILED(m_pSourceVoice->Start(0)))
+        {
+            Log::Print("Error playing audio\n");
+        }
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp]: If this is a print to warn the developer that the sound has not successfully started playing, please make it a warning insted.

File: StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp [hunk 3]
```diff
@@ -79,3 +82,37 @@ void AudioVoice::Stop()
...
+
+void AudioVoice::Pause()
+{
+    if (m_Initialized)
+    {
+        // stop playback
+        if (FAILED(m_pSourceVoice->Stop(0)))
+        {
+            Log::Print("Error stopping audio\n");
+        }
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp]: If this is a print to warn the developer that the sound has not successfully stopped playing, please make it a warning insted.

File: StortSpelProjekt/Engine/src/ECS/Components/AnimationComponent.cpp [hunk 4]
```diff
@@ -72,19 +72,22 @@ void component::AnimationComponent::walkAnimation(MovementInput* evnt)
...
-	else
-	{
-	}
+		if (evnt->button == MOUSE_BUTTON::LEFT_DOWN)
+		{
+			m_pAnimatedModel->PlayAnimation("Attack_Swing_Right", false);
+		}
+		else
+		{
+		}
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/ECS/Components/AnimationComponent.cpp]: Why? Please remove this

File: StortSpelProjekt/Engine/src/ECS/Components/AnimationComponent.cpp [hunk 5]
```diff
@@ -72,19 +72,22 @@ void component::AnimationComponent::walkAnimation(MovementInput* evnt)
...
-	}
-	else
-	{
+		if (evnt->button == MOUSE_BUTTON::RIGHT_DOWN)
+		{
+			m_pAnimatedModel->PlayAnimation("Attack_Swing_Left", false);
+		}
+		else
+		{
+		}
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/ECS/Components/AnimationComponent.cpp]: Why? Please remove this

File: StortSpelProjekt/Engine/src/ECS/Components/Audio2DVoiceComponent.cpp [hunk 6]
```diff
@@ -81,3 +83,31 @@ void component::Audio2DVoiceComponent::Stop(const std::wstring& name)
...
 }
+
+void component::Audio2DVoiceComponent::pauseToggleAudio(PauseGame* evnt)
+{
+	if (evnt->isPaused)
+	{
+		for (auto voice : m_Voices)
+		{
+			XAUDIO2_VOICE_STATE test;
+			voice.second.GetSourceVoice()->GetState(&test);
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/ECS/Components/Audio2DVoiceComponent.cpp]: As before, please change the temporary variable name from test to something more descriptive. voiceState perhaps?

File: StortSpelProjekt/Engine/src/ECS/Components/Audio3DEmitterComponent.cpp [hunk 7]
```diff
@@ -156,3 +159,31 @@ void component::Audio3DEmitterComponent::Stop(const std::wstring& name)
...
 }
+
+void component::Audio3DEmitterComponent::pauseToggleAudio(PauseGame* evnt)
+{
+	if (evnt->isPaused)
+	{
+		for (auto voice : m_VoiceEmitterData)
+		{
+			XAUDIO2_VOICE_STATE test;
+			voice.second.voice.GetSourceVoice()->GetState(&test);
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/ECS/Components/Audio3DEmitterComponent.cpp]: Please change this variable name

File: StortSpelProjekt/Vendor/Resources/Models/BannerHans/banner.mtl [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+Kd 0.215861 0.215861 0.215861
+Ks 1.000000 1.000000 1.000000
+Ke 0.000000 0.000000 0.000000
+Ni 1.450000
+d 1.000000
+illum 3
+map_Kd albedo.DDS
+map_Ka metallic.DDS
+map_Ks roughness.DDS
+map_Kn normal.DDS
```
Comment (Schtekt) [inline StortSpelProjekt/Vendor/Resources/Models/BannerHans/banner.mtl:16]: Nice

File: StortSpelProjekt/Vendor/Resources/Models/BannerStefan/banner.mtl [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+Kd 0.215861 0.215861 0.215861
+Ks 1.000000 1.000000 1.000000
+Ke 0.000000 0.000000 0.000000
+Ni 1.450000
+d 1.000000
+illum 3
+map_Kd albedo.DDS
+map_Ka metallic.DDS
+map_Ks roughness.DDS
+map_Kn normal.DDS
```
Comment (Schtekt) [inline StortSpelProjekt/Vendor/Resources/Models/BannerStefan/banner.mtl:16]: Nice

**Review by Schtekt — CHANGES_REQUESTED** [review]
In general, I love it!
I found some small things that I want to see changed.

**Review by Schtekt — APPROVED** [review]
Great work!

**Review by AntonAsbrink — APPROVED** [review]

**Pull request merged**